### PR TITLE
update distributions with custom header

### PR DESCRIPTION
**Pull request by scytacki**
This code has already been run on the production account. It is useful to check in and review so other developers know these scripts exist for when we have to change the buckets or distributions again.

More information about Buckets, Distributions, and CORS can be found here: https://docs.google.com/document/d/1TJxY5hYMgetRj-YQn06YuichDQP9lQINELtrD7MyjT4/edit#

This PR addresses the "New issue not handled by current setup" paragraph. That document needs to be updated to reflect this change.

The specific change this code applies is to add a custom `Origin: https://concord.org` header to most of the Distribution Origins (2 different things called origin here). It isn't applied to every Distribution Origin though since some Origins point to servers not S3 buckets.

### CODE REVIEW

File: s3-cloudfront-scripts/README.md [hunk 1]
```diff
@@ -63,18 +64,27 @@ Generate a `behaviors.csv` with information about the behaviors of all of the Cl
 
 ## cf-behavior-list (update enabled)
 
-If you uncomment the last line in the `cf-behavior-list.mjs` file then the script will also update the distributions. This was used to update all behaviors using public S3 buckets so they would use the S3-CORS cache policy.
+At the end of `cf-behavior-list.mjs` are a 2 sections you can uncomment to update the distributions. The first was used to update all behaviors using public S3 buckets so they would use the S3-CORS cache policy. The second was used to update all origins pointing at public S3 buckets so they'd have a custom header of `Origin: https://concord.org`
```
Comment (kswenson) [inline s3-cloudfront-scripts/README.md:67]:
```suggestion
At the end of `cf-behavior-list.mjs` are two sections you can uncomment to update the distributions. The first was used to update all behaviors using public S3 buckets so they would use the S3-CORS cache policy. The second was used to update all origins pointing at public S3 buckets so they'd have a custom header of `Origin: https://concord.org`
```

File: s3-cloudfront-scripts/README.md [hunk 2]
```diff
@@ -63,18 +64,27 @@ Generate a `behaviors.csv` with information about the behaviors of all of the Cl
...
 
-The update script part of the script looks for distributions meeting the following criteria. It has to have at least one behavior that has all of the following properties:
+These sections looks for distributions meeting the certain criteria. Then for all matching distributions are passed to `updateDistributions` along with an option `modifyBehavior` and `modifyOrigin` function.
```
Comment (kswenson) [inline s3-cloudfront-scripts/README.md:69]:
```suggestion
These sections looks for distributions meeting certain criteria. Then all matching distributions are passed to `updateDistributions` along with an option `modifyBehavior` and `modifyOrigin` function.
```

File: s3-cloudfront-scripts/README.md [hunk 3]
```diff
@@ -63,18 +64,27 @@ Generate a `behaviors.csv` with information about the behaviors of all of the Cl
...
+
+The update code is written to be modular so you can use these sections as a guide and add your own code for updating all of the distributions. 
+
+It is useful to test these modification with a single distribution before applying the changes to all distributions. See the test-update-behavior for that.
```
Comment (kswenson) [inline s3-cloudfront-scripts/README.md:73]:
```suggestion
It is useful to test these modifications with a single distribution before applying the changes to all distributions. See the test-update-behavior for that.
```

File: s3-cloudfront-scripts/README.md [hunk 4]
```diff
@@ -63,18 +64,27 @@ Generate a `behaviors.csv` with information about the behaviors of all of the Cl
...
+Each matching distribution has each of behavior that meets the criteria above updated to use the S3-CORS cache policy. Cache Policies can only be associated with 100 distributions. So the script works with both the S3-CORS and S3-CORS-1 policies.
+
+### Custom Header Update details
+For this section, we look for distributions that don't already have a custom header of Origin, and that have a bucket origin that has a public or public-custom-cors tag.
 
-The update code is written to be modular so you can replace the `modifyBehavior` function to look for different conditions and/or update the behavior differently. But note that that thre is a separate filter to figure out which distributions need updating, so that filter needs to be updated too.
+Then we update each matching origin with the new custom header. There are some distributions that have multiple origins, so we need to check which origins need update in additional to figuring out which distributions need updating.
```
Comment (kswenson) [inline s3-cloudfront-scripts/README.md:86]:
```suggestion
Then we update each matching origin with the new custom header. There are some distributions that have multiple origins, so we need to check which origins need updating in addition to figuring out which distributions need updating.
```

**Comment by scytacki** [comment]
I think @kswenson's review is good enough. At least one other person is aware of this stuff. ;)

**Pull request merged**